ui/src/me/components/account/ViewTokenOverlay.tsx

```diff
@@ -15,21 +15,26 @@ import {Authorization, Permission} from '@influxdata/influx'
 // Actions
 import {NotificationAction} from 'src/types'
 
-const {Write, Read} = Permission.ActionEnum
-
 interface Props {
   onNotify: NotificationAction
   auth: Authorization
   onDismissOverlay: () => void
 }
 
-const actions = [Read, Write]
-
 export default class ViewTokenOverlay extends PureComponent<Props> {
   public render() {
     const {description, permissions} = this.props.auth
     const {onNotify} = this.props
 
+    const obj = {}
```

Comment:
I often see objects like this called something like `permissionsByType`, letting you know that it's an object whose keys are types and values are permissions.

A more functional style of building up that object could go something like:
```
permissions.reduce((acc, {action, resource: {type}) => ({
  ...acc,
  [type]: [...acc[type], action]
}), {})
```
Whether or not that's more readable is not so certain :)

Reply:
Thanks!